src/kzg10/subproductdomain.rs

```diff
@@ -0,0 +1,424 @@
+use crate::Error;
+use ark_ec::PairingEngine;
+use ark_ff::{FftField, Field, Zero};
+use ark_poly::{Polynomial, UVPolynomial};
+
+use ark_poly::polynomial::univariate::DensePolynomial as Poly;
+
+/// Where indicated, algorithms are from Modern Computer Algebra, 3rd edition, by Gathen and Gerhard
+/// Abbreviated as GG
+/// Let M(n) denote the time to multiply.
```

Comment:
Maybe this should become module level documentation? So 
```suggestion
//! Where indicated, algorithms are from Modern Computer Algebra, 3rd edition, by Gathen and Gerhard
//! Abbreviated as GG
//! Let M(n) denote the time to multiply.
```